.spin/cmds.py

```diff
@@ -142,16 +142,46 @@ def test(ctx, pytest_args, markexpr, n_jobs, tests, verbose):
 
 
 @click.command()
-@click.argument('python_expr')
-def gdb(python_expr):
+@click.option('--code', '-c', help='Python program passed in as a string')
+@click.argument('gdb_args', nargs=-1)
+def gdb(code, gdb_args):
     """👾 Execute a Python snippet with GDB
 
+    spin gdb -c 'import numpy as np; print(np.__version__)'
+
+    Or pass arguments to gdb:
+
+    spin gdb -c 'import numpy as np; print(np.__version__)' -- --fullname
+
+    Or run another program, they way you normally would with gdb:
+
+    spin gdb ls
+    spin gdb -- --args ls -al
+
+    You can also run Python progreams:
+
+    spin gdb my_tests.py
+    spin gdb -- my_tests.py --mytest-flag
     """
-    util.run(
-        ['gdb', '--args', 'python', '-m', 'spin', 'run',
-         'python', '-P', '-c', python_expr],
-        replace=True
-    )
+    meson._set_pythonpath()
+    gdb_args = list(gdb_args)
+
+    if gdb_args and gdb_args[0].endswith('.py'):
+        gdb_args = ['--args', sys.executable] + gdb_args
+
+    if sys.version_info[:2] >= (3, 11):
+        PYTHON_FLAGS = ['-P']
+        code_prefix = ''
+    else:
+        PYTHON_FLAGS = []
+        code_prefix = 'import sys; sys.path.pop(0); '
+
+    if code:
+        PYTHON_ARGS = ['-c', code_prefix + code]
+        gdb_args += ['--args', sys.executable] + PYTHON_FLAGS + PYTHON_ARGS
+
+    gdb_cmd = ['gdb', '-ex', 'set follow-fork-mode child'] + gdb_args
```

Comment:
```suggestion
    gdb_cmd = ['gdb', '-ex', 'set detach-on-fork on'] + gdb_args
```

When I suggested `follow-fork-mode child` earlier I was testing with a script that didn't import numpy. Actually importing numpy spawns some threads which seems to interfere with `follow-fork-mode child`. It seems we actually want to debug both the child and parent process.